Library/Homebrew/tap.rb

```diff
@@ -872,26 +875,25 @@ def ==(other)
     self.class == other.class && name == other.name
   end
 
-  def self.each(&block)
-    return to_enum unless block
+  # A set with all installed and core taps.
+  #
+  # @private
+  sig { returns(T::Set[Tap]) }
```

Comment:
We might need to implement `#hash` and `eql?` to make the set-like properties work here. Worth checking the following scenario.

```rb
pp Tap.tap_registry
Tap.new("Homebrew/portable-ruby").install
pp Tap.tap_registry
Tap.new("Homebrew/portable-ruby").uninstall
pp Tap.tap_registry
```

`Tap.fetch` avoids this problem because of the cache but still.

Reply:
I'd overlooked that because as you mention `Tap.fetch` already does most of the heavy lifting here.